tools/bin/hawqregister

```diff
@@ -325,250 +326,221 @@ class HawqRegister(object):
-                    sys.exit(1)
+    def check_hash_type(self):
+        self.accessor.check_hash_type(self.dst_table_name)
 
-        def check_database_encoding():
-            encoding_indx = self.accessor.get_database_encoding_indx(self.database)
-            encoding = self.accessor.get_database_encoding(encoding_indx)
-            if self.encoding.strip() != encoding:
-                logger.error('Database encoding from yaml configuration file(%s) is not consistent with encoding from input args(%s).' % (self.encoding, encoding))
-                sys.exit(1)
-
-        def create_table():
-            try:
-                (ret, query) = self.accessor.do_create_table(self.src_table_name, self.tablename, self.schema, self.file_format, self.distribution_policy, self.file_locations, self.bucket_number,
+    def create_table(self):
+        try:
+           (ret, query) = self.accessor.do_create_table(self.src_table_name, self.dst_table_name, self.schema, self.file_format, self.distribution_policy, self.file_locations, self.bucket_number,
                                                              self.partitionby, self.partitions_constraint, self.partitions_name)
-            except pg.DatabaseError as e:
-                print e
+        except pg.DatabaseError as e:
+            print e
+            sys.exit(1)
+        if ret:
+            self.failure_handler.commit(('SQL', query))
+        return ret
+
+    def check_database_encoding(self):
+        encoding_indx = self.accessor.get_database_encoding_indx(self.database)
+        encoding = self.accessor.get_database_encoding(encoding_indx)
+        if self.encoding.strip() != encoding:
+            logger.error('Database encoding from yaml configuration file(%s) is not consistent with encoding from input args(%s).' % (self.encoding, encoding))
+            sys.exit(1)
+
+    def check_policy_consistency(self):
+        policy = self.get_distribution_policy() # "" or "{1,3}"
+        if policy is None:
+            return
+        if self.distribution_policy == 'DISTRIBUTED RANDOMLY':
+            logger.error('Distribution policy of %s from yaml is not consistent with the policy of existing table.' % self.tablename)
+            self.failure_handler.rollback()
+            sys.exit(1)
+        tmp_dict = {}
+        for i, d in enumerate(self.schema):
+            tmp_dict[d['name']] = i + 1
+        # 'DISTRIBUETD BY (1,3)' -> {1,3}
+        cols = self.distribution_policy.strip().split()[-1].strip('(').strip(')').split(',')
+        original_policy = ','.join([str(tmp_dict[col]) for col in cols])
+        if policy.strip('{').strip('}') != original_policy:
+            logger.error('Distribution policy of %s from yaml file is not consistent with the policy of existing table.' % self.dst_table_name)
+            self.failure_handler.rollback()
+            sys.exit(1)
+
+    def set_yml_dataa(self, file_format, files, sizes, tablename, schema, distribution_policy, file_locations,\
```

Comment:
Why uses set_yml_dataa not set_yml_data? What does the additional "a" stand for?